bin/zowe-verify-authenticity.sh

```diff
@@ -0,0 +1,374 @@
+
+echo `date`                         >> $LOG_FILE
+echo "runtimePath = $runtimePath"   >> $LOG_FILE
+echo "hashPath    = $hashPath"      >> $LOG_FILE
+echo "refPath     = $refPath"       >> $LOG_FILE
+echo "outputPath  = $outputPath"    >> $LOG_FILE
+
+# - - - - - - - - - - - - - main logic
+
+# Verify runtime directory contents minimally
+for dir in bin components scripts manifest.json
+do
+    ls -l $runtimePath | grep " $dir$" 1> /dev/null 2>/dev/null
+    if [[ $? -ne 0 ]]
+    then
+        echo Error: runtimePath $runtimePath does not contain $dir | tee -a $LOG_FILE
+        exit 1
+    fi 
+done 
+
+# Reference hash file will be named like RefRuntimeHash-1.12.0.txt
+# manifest.json will contain a line like this near the top:
+#   "version": "1.11.0",
+# Extract Zowe version from manifest.json: 
+
+zoweVersion=`head $runtimePath/manifest.json | sed -n 's/ *"version" *: *"\(.*\)".*/\1/p'`
+if [[ -n $zoweVersion ]]
+then 
+    echo Info: zoweVersion = $zoweVersion | tee -a $LOG_FILE
+else
+    echo Error: Unable to obtain zoweVersion from $runtimePath/manifest.json | tee -a $LOG_FILE
+    exit 1
+fi
+
+# Verify hash file directory contents minimally
+if [[ ! -f $hashPath/HashFiles.class ]]
+then
+    echo Error: hashPath $hashPath does not contain HashFiles.class file | tee -a $LOG_FILE
+    exit 1
+fi 
+
+RefRuntimeHash=RefRuntimeHash-$zoweVersion.txt
+# Verify refPath directory contents minimally
+if [[ ! -f $refPath/$RefRuntimeHash ]]
+then
+    echo Error: refPath $refPath does not contain $RefRuntimeHash | tee -a $LOG_FILE
+    exit 1
+fi
+
+# Verify refPath contents minimally
+head $refPath/$RefRuntimeHash|grep '^\./' 1> /dev/null 2>/dev/null
+if [[ $? -ne 0 ]]
+then
+    echo Error: Lines at top of file $refPath/$RefRuntimeHash do not start with \"./\" | tee -a $LOG_FILE
+    exit 1
+fi 
+
+echo Info: Gathering files ... | tee -a $LOG_FILE
+
+cd $runtimePath
+. bin/internal/zowe-set-env.sh # ensure we have tagging behaviour set correctly
+
+find . -name ./SMPE          -prune \
+    -o -name "./ZWE*"        -prune \
+    -o -name ./fingerprint   -prune \
+    -o -type f -print > $outputPath/files.in # exclude SMPE, ZWE* and fingerprint
+if [[ $? -ne 0 ]]
+then
+    echo Error: Failed to generate a list of files from $runtimePath | tee -a $LOG_FILE
+    exit 1
+fi 
+
+echo Info: Checking java version >> $LOG_FILE
+java -version 2>> $LOG_FILE
+if [[ $? -ne 0 ]]
+then
+    echo "Warning: java not in PATH" | tee -a $LOG_FILE
+    if [[ -n "$JAVA_HOME" ]]
+    then
+        echo Info: JAVA_HOME = $JAVA_HOME  >> $LOG_FILE
+        $JAVA_HOME/bin/java -version 2>> $LOG_FILE
+        if [[ $? -ne 0 ]]
+        then
+            echo "Error: Cannot find java version in $JAVA_HOME/bin" | tee -a $LOG_FILE
+            exit 1
+        else
+            javaPrefix=$JAVA_HOME/bin/
+        fi
+    else
+        echo "Error: JAVA_HOME is not set" | tee -a $LOG_FILE
+        exit 1
+    fi
+else
+    javaPrefix=  # java is in $PATH
+fi
```

Comment:
Please use the existing java-utils.sh scripts for this rather than duplicating

Reply:
Done.